codegen/src/serialization.rs

```diff
@@ -0,0 +1,87 @@
+use std::collections::BTreeMap;
```

Comment:
Is the purpose of this whole file to provide Serde with an override so it knows how to handle shape names that need to be deserialized with `capitalize_first` called on them?